### DISCUSSION TITLE
Dynamic unix_domains

### DISCUSSION THREAD

**Discussion by rygwdn** [wezterm/wezterm#5581] [Q&A]
### What Operating System(s) are you running on?

macOS

### WezTerm version

20240617-083535-cb140686

### Ask your question!

I would like to be able to connect to multiple hosts using the same wezterm instance. I am able to do this pretty well with unix_domains and proxy_command (I can't use ssh_domains because I need to use a wrapper program to access the hosts), but it seems that all of the domains _must_ be configured before wezterm is started. Is this right? I connect to brand new hosts pretty frequently (2-3x per day), so restarting wezterm each time is pretty annoying!

I've tried dynamically building the unix_domains value and reloading my config, but it seems that the domains don't get updated on refresh. Is there any other way to accomplish this without restarting wezterm every time?

**Comment by wez** [comment]
It's not currently supported. https://github.com/wez/wezterm/issues/2540 is effectively the same issue.

As an alternative/approximation, you might consider launching a new wezterm via `wezterm --config 'unix_domains={{name='adhoc', ..}' connect adhoc` to define the domain for the client inline as a command line parameter, presumably also with your bootstrap/proxy command; that way you don't have to write it out into a config file.

What you can't do with this is have the newly spawned client be a tab or pane in your main wezterm gui.

**Reply by rygwdn** [reply]
Thanks! I had considered that, but I'd really like to have them in the same UI. Is this something that would be a major change to the code? If it's a smallish change, I might be able to find time to get something working if you'd be interested in a PR.

**Reply by wez** [reply]
It's largish to do the way that I have in mind; I actually have a local branch started that will facilitate this feature, among a couple of other things, I've just had a lot of stuff going on and not as much time to make progress on it as I'd have liked.
